app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Campaign;
+use App\Models\User;
+use App\Models\Character;
+use Illuminate\Support\Facades\Auth;
+use Inertia\Inertia;
+
+class CampaignRegistrationController extends Controller
+{
+    /**
+     * @param \Illuminate\Http\Request $request
+     * @return \Inertia\Response
+     */
+    public function create(Request $request)
+    {
+        $characters = Auth::user()->characters;
+
+        $code = $request->validate([
+            'code' => "sometimes|string"
+        ])['code']?? null;
+
+        $campaign = Campaign::where('code', $code)->first();
+
+        $code = is_null($campaign) ? $campaign : $campaign->code;
+
+        return Inertia::render('Campaign/Detail/CampaignDetail', compact('characters', 'campaign', 'code'));
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'character_id' => "sometimes|exists:characters,id",
```

Comment:
knowing the front ends, its likely they're going to pass `"character_id": null`, so might as well just add 
```suggestion
            'character_id' => "nullable|sometimes|exists:characters,id",
```